workflow/controller/dag.go

```diff
@@ -358,6 +358,22 @@ func (woc *wfOperationCtx) executeDAGTask(ctx context.Context, dagCtx *dagContex
 
 	node := dagCtx.getTaskNode(taskName)
 	task := dagCtx.GetTask(taskName)
+	if node != nil && (node.Fulfilled() || node.Phase == "Running") {
+		scope, err := woc.buildLocalScopeFromTask(dagCtx, task)
+		if err != nil {
+			woc.markNodeError(node.Name, err)
+		}
+		scope.addParamToScope(fmt.Sprintf("tasks.%s.status", task.Name), string(node.Phase))
```

Comment:
not your mistake, but possible null pointer here. You should log and then return after the markNodeError

Reply:
I moved the code as it was, but I couldn't check that a null pointer could occur. Thank you for the detailed review.

Reply:
Ah so `scope` can be a nil pointer I believe. So in the err != nil branch, just add a log and return. 
This was a bug that was present before your changes though, but its probably fine to make the change here. 
Alternatively I can push something too in a different branch